### ISSUE TITLE
GetAtt logic when going to an Object

### ISSUE THREAD

**Issue by kddejong** [aws-cloudformation/cfn-lint#3476]
### CloudFormation Lint Version

>1

### What operating system are you using?

Mac

### Describe the bug

Currently when we have a readOnlyProperties pointer to `/properties/Outputs` that looks like
```json
  "Outputs": {
   "additionalProperties": false,
   "maxProperties": 100,
   "patternProperties": {
    "^[A-Za-z0-9]{1,64}$": {
     "$ref": "#/definitions/OutputType"
    }
   },
   "type": "object"
  }
```

We aren't building out the GetAtt to be `Outputs\\..*` plus we assume the pointers go to properties.  We need to figure out will work for the pointer and how to build the GetAtts for this.

Related issue #3474

### Expected behavior

That we build the GetAtt options correctly.  

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: Serverless Stack
Resources:
  SimpleLambda:
    Type: AWS::ServiceCatalog::CloudFormationProvisionedProduct
    Properties:
      ProductName: Basic Lambda
      ProvisioningArtifactName: '1.0'
            
  SimpleApiGateway:
    Type: AWS::ServiceCatalog::CloudFormationProvisionedProduct
    Properties:
      ProductName: API Gateway
      ProvisioningArtifactName: '1.0'
      ProvisioningParameters:
        - Key: LambdaArn
          Value: !GetAtt [SimpleLambda, Outputs.SCLambdaArn]
```
